### PR TITLE
Added commands for gifs

### PR DESCRIPTION
**Pull request by BlackDemonFire**
cry, cuddle, hug, kiss and pat
added "gifs" Enmap to clientProperties
added random_choice function to get a random element from an array

### CODE REVIEW

**Comment by Chicken** [comment]
ffs I just manually added the hug command because you were taking so long

File: src/commands/Gifs/cry.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+let discord = require("discord.js");
```
Comment (Chicken) [inline src/commands/Gifs/cry.js:1]: Discord is capitalized everywhere else in the project / naming convention

File: src/commands/Gifs/cry.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+// eslint-disable-next-line no-unused-vars
+exports.execute = async (client, message, args) => {
+    let gif = await client.random_choice(client.gifs.get("cry"));
```
Comment (Chicken) [inline src/commands/Gifs/cry.js:4]: random_choice is not an async function nor returns a promise, doesn't need await

File: src/commands/Gifs/cuddle.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+let discord = require("discord.js");
+exports.execute = async (client, message, args) => {
+    let gif = await client.random_choice(client.gifs.get("cuddle"));
+    let embed = new discord.RichEmbed()
```
Comment (Chicken) [inline src/commands/Gifs/cuddle.js:4]: should be MessageEmbed

File: src/commands/Gifs/hug.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+let discord = require("discord.js");
+exports.execute = async (client, message, args) => {
+    let gif = await client.random_choice(client.gifs.get("hug"));
+    let embed = new discord.RichEmbed()
```
Comment (Chicken) [inline src/commands/Gifs/hug.js:4]: MessageEmbed

File: src/commands/Gifs/kiss.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+let discord = require("discord.js");
+exports.execute = async (client, message, args) => {
+    let gif = await client.random_choice(client.gifs.get("kiss"));
+    let embed = new discord.RichEmbed()
```
Comment (Chicken) [inline src/commands/Gifs/kiss.js:4]: MessageEmbed

File: src/functions/clientProperties.js [hunk 6]
```diff
@@ -11,4 +11,5 @@ module.exports = async client => {
     client.music = new Enmap("music");
     client.bantimers = {};
     client.remindtimers = {};
+    client.gifs = new Enmap("gifs");
```
Comment (Chicken) [inline src/functions/clientProperties.js:14]: Tbh I think doing this in an enmap is a bad idea. This way the urls are not on the github repo. JSON file might be a better solution.

File: src/functions/util.js [hunk 7]
```diff
@@ -124,4 +124,8 @@ module.exports = async client => {
     process.on("exit", async code => {
         client.logger.error("Script exited with code " + code);
     });
+    client.random_choice = (arr) => {
```
Comment (Chicken) [inline src/functions/util.js:127]: randomChoice, we use lower camel case, even better would be "arrayRandom"

File: src/commands/Gifs/cuddle.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+let discord = require("discord.js");
```
Comment (Chicken) [inline src/commands/Gifs/cuddle.js:1]: Discord, naming convention

File: src/commands/Gifs/cuddle.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+    let name;
+    let cuddled;
+    let outstr = args.join(" ");
+    if (message.member) {
+        name = message.member.displayName;
+    } else {
+        name = message.author.username;
+    }
+    if (args.length==1&&message.mentions.users.length==1) cuddled = message.mentions.users.first().username;
```
Comment (Chicken) [inline src/commands/Gifs/cuddle.js:16]: could you add spaces in places like this

File: src/commands/Gifs/hug.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
+let discord = require("discord.js");
```
Comment (Chicken) [inline src/commands/Gifs/hug.js:1]: Discord, naming convention

File: src/commands/Gifs/kiss.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
+let discord = require("discord.js");
```
Comment (Chicken) [inline src/commands/Gifs/kiss.js:1]: Discord, naming convention

File: src/commands/Gifs/pat.js [hunk 12]
```diff
@@ -0,0 +1,31 @@
+let discord = require("discord.js");
```
Comment (Chicken) [inline src/commands/Gifs/pat.js:1]: Discord, naming convention

**Review by Chicken — CHANGES_REQUESTED** [review]
Also, pull the changes from main repo and remove the hug command and json file I created

File: src/commands/Gifs/pat.js [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+exports.execute = async (client, message, args) => {
+    let gif = await client.random_choice(client.gifs.get("pat"));
+    let embed = new discord.RichEmbed()
```
Comment (WilsontheWolf) [inline src/commands/Gifs/pat.js:4]: Should also be MessageEmbed

File: src/commands/Gifs/cuddle.js (hunk 9 above)
Comment (WilsontheWolf) [inline src/commands/Gifs/cuddle.js:16]:
```suggestion
    if (args.length == 1 && message.mentions.users.length == 1) cuddled = message.mentions.users.first().username;
```

File: src/commands/Gifs/hug.js [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+    if (message.member) {
+        name = message.member.displayName;
+    } else {
+        name = message.author.username;
+    }
+    if (args.length==1&&message.mentions.users.length==1) hugged = message.mentions.users.first().username;
+    if (hugged) outstr = `${name} hugs ${hugged}`;
+    if (!outstr||outstr=="") outstr = `${name} needs hugs ...`;
+    embed.setColor("RANDOM");
+    embed.setDescription(outstr);
```
Comment (WilsontheWolf) [inline src/commands/Gifs/hug.js:19]: keeping the embed stuff to one chain is a bit cleaner.

File: src/commands/Gifs/hug.js [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+        .setImage(gif);
+    let name;
+    let hugged;
+    let outstr = args.join(" ");
+    if (message.member) {
+        name = message.member.displayName;
+    } else {
+        name = message.author.username;
+    }
+    if (args.length==1&&message.mentions.users.length==1) hugged = message.mentions.users.first().username;
```
Comment (WilsontheWolf) [inline src/commands/Gifs/hug.js:15]: spacing
```suggestion
    if (args.length == 1 && message.mentions.users.length == 1) hugged = message.mentions.users.first().username;
```

File: src/commands/Gifs/kiss.js [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+        .setImage(gif);
+    let name;
+    let kissed;
+    let outstr = args.join(" ");
+    if (message.member) {
+        name = message.member.displayName;
+    } else {
+        name = message.author.username;
+    }
+    if (args.length==1&&message.mentions.users.length==1) kissed = message.mentions.users.first().username;
```
Comment (WilsontheWolf) [inline src/commands/Gifs/kiss.js:15]:
```suggestion
    if (args.length == 1 && message.mentions.users.length == 1) kissed = message.mentions.users.first().username;
```

File: src/commands/Gifs/pat.js [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+        .setImage(gif);
+    let name;
+    let patted;
+    let outstr = args.join(" ");
+    if (message.member) {
+        name = message.member.displayName;
+    } else {
+        name = message.author.username;
+    }
+    if (args.length==1&&message.mentions.users.length==1) patted = message.mentions.users.first().username;
```
Comment (WilsontheWolf) [inline src/commands/Gifs/pat.js:15]:
```suggestion
    if (args.length == 1 && message.mentions.users.length == 1) patted = message.mentions.users.first().username;
```

File: src/commands/Gifs/pat.js [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+    if (patted) outstr = `${name} pats ${patted}`;
+    if (!outstr||outstr=="") outstr = `${name} needs headpats ...`;
```
Comment (WilsontheWolf) [inline src/commands/Gifs/pat.js:17]:
```suggestion
    if (!outstr || outstr == "") outstr = `${name} needs headpats ...`;
```

File: src/commands/Gifs/kiss.js [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+    if (kissed) outstr = `${name} kisses ${kissed} OwO`;
+    if (!outstr||outstr=="") outstr = `${name} needs kisses UwU`;
```
Comment (WilsontheWolf) [inline src/commands/Gifs/kiss.js:17]:
```suggestion
    if (!outstr || outstr == "") outstr = `${name} needs kisses UwU`;
```

File: src/commands/Gifs/hug.js [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+    if (hugged) outstr = `${name} hugs ${hugged}`;
+    if (!outstr||outstr=="") outstr = `${name} needs hugs ...`;
```
Comment (WilsontheWolf) [inline src/commands/Gifs/hug.js:17]:
```suggestion
    if (!outstr || outstr == "") outstr = `${name} needs hugs ...`;
```

File: src/functions/util.js [hunk 21]
```diff
@@ -124,4 +124,8 @@ module.exports = async client => {
...
+        if (arr.length==0) return arr;
```
Comment (WilsontheWolf) [inline src/functions/util.js:128]:
```suggestion
        if (arr.length == 0) return arr;
```

File: src/commands/Gifs/cuddle.js [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+    if (cuddled) outstr = `${name} cuddles ${cuddled}`;
+    if (!outstr||outstr=="") outstr = `${name} needs cuddles ...`;
```
Comment (WilsontheWolf) [inline src/commands/Gifs/cuddle.js:18]:
```suggestion
    if (!outstr || outstr == "") outstr = `${name} needs cuddles ...`;
```

**Review by WilsontheWolf — CHANGES_REQUESTED** [review]
Mostly spacing. 
Also a rich embed chicken missed

File: src/functions/util.js [hunk 23]
```diff
@@ -124,4 +124,8 @@ module.exports = async client => {
     process.on("exit", async code => {
         client.logger.error("Script exited with code " + code);
     });
+    client.arrayRandom = (arr) => {
+        if (arr.length == 0) return arr;
```
Comment (Chicken) [inline src/functions/util.js:128]:
```suggestion
        if (!arr && arr.length == 0) return null;
```

**Review by Chicken — CHANGES_REQUESTED** [review]

**Review by Chicken — APPROVED** [review]

**Review by WilsontheWolf — APPROVED** [review]
Looks good

**Pull request merged**